tests/notifier/notifiers/email_violations_test.py

```diff
@@ -0,0 +1,116 @@
+                 '"CAE=", "kind": "storage#policy", "resourceId": "projects'
+                 '/_/buckets/be-1-ext"}'),
+             'resource_id': 'be-1-ext',
+             'resource_type': 'bucket',
+             'rule_index': 1,
+             'rule_name': 'Allow only service accounts to have access',
+             'violation_data': {'full_name': 'o/5/f/4/f/9/p/be-p1-196611/bucket/be-1-ext/',
+                                 'member': u'user:abc@example.com',
+                                 'role': u'roles/storage.objectAdmin'},
+             'violation_type': 'ADDED'},
+            {'full_name': 'o/5/f/4/f/9/p/be-p1-196611/bucket/be-1-int/',
+             'inventory_data': (
+                 '{"bindings": [{"members": ["projectEditor:be-p1-196611", '
+                 '"projectOwner:be-p1-196611"], "role": "roles/storage.lega'
+                 'cyBucketOwner"}, {"members": ["projectViewer:be-p1-196611'
+                 '"], "role": "roles/storage.legacyBucketReader"}], "etag":'
+                 '"CAE=", "kind": "storage#policy", "resourceId": "projects'
+                 '/_/buckets/be-1-int"}'),
+             'resource_id': 'be-1-int',
+             'resource_type': 'bucket',
+             'rule_index': 1,
+             'rule_name': 'Allow only service accounts to have access',
+             'violation_data': {'full_name': 'o/5/f/4/f/9/p/be-p1-196611/bucket/be-1-int/',
+                                'member': u'user:ab.cd@example.com',
+                                'role': u'roles/storage.objectViewer'},
+             'violation_type': 'ADDED'}]
+        self.evp = email_violations.EmailViolations(
+            'policy_violations',
+            '2018-03-14T14:49:36.101287',
+            self.violations,
+            fake_global_conf,
+            {},
+            fake_pipeline_conf)
+        self.test_data_path = os.path.join(
+                os.path.dirname(__file__), 'test_data_attachment.csv')
+
+    @mock.patch(
+        'google.cloud.forseti.notifier.notifiers.email_violations.date_time',
+        autospec=True)
+    def test_get_output_filename(self, mock_date_time):
+        """Test _get_output_filename()."""
+        mock_date_time.get_utc_now_datetime = mock.MagicMock()
+        mock_date_time.get_utc_now_datetime.return_value = self.fake_utcnow
+        expected_timestamp = self.fake_utcnow.strftime(
+            string_formats.TIMESTAMP_TIMEZONE_FILES)
+
+        actual_filename = self.evp._get_output_filename()
+        self.assertEquals(
+            string_formats.VIOLATION_CSV_FMT.format(
+                self.evp.resource, self.evp.cycle_timestamp,
+                expected_timestamp),
+            actual_filename)
+
+    def test_write_temp_attachment(self):
+        """Test _write_temp_attachment()."""
+        file_name = self.evp._write_temp_attachment()
+        self.assertTrue(
+            filecmp.cmp('/tmp/%s' % file_name,
+                        self.test_data_path, shallow=False))
```

Comment:
When we are here, it's hard to tell what `test_data_path` is.  It would be better to call this `self.expected_attachment_path`, for example.  Then, it's super clear, what we are comparing.

Reply:
Done.